### PR TITLE
Test: Create KV test driver with callback. Add test for nil interface

### PR DESCRIPTION
**Pull request by N-o-Z**

### CODE REVIEW

File: pkg/kv/testdriver/store.go [hunk 1]
```diff
@@ -0,0 +1,112 @@
+package testdriver
```
Comment (itaiad200) [inline pkg/kv/testdriver/store.go:1]: I think you reimplemented the code of mockgen, it gives you exactly what you need here.

**Review by itaiad200 — CHANGES_REQUESTED** [review]
I suggest giving mockgen a spin before creating our own mock-with-callbacks implementations.

**Review by itaiad200 — CHANGES_REQUESTED** [review]
I suggest giving mockgen a spin before creating our own mock-with-callbacks implementations.

File: pkg/kv/kvtest/store.go [hunk 2]
```diff
@@ -869,5 +870,5 @@ func GetStore(ctx context.Context, t testing.TB) kv.Store {
 	t.Cleanup(func() {
 		store.Close()
 	})
-	return store
+	return testdriver.NewTestStore(store)
```
Comment (eden-ohana) [inline pkg/kv/kvtest/store.go:873]: why not test the actual store?

File: pkg/kv/iterators_test.go [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+	err := sm.SetMsg(ctx, modelPartitionKey, setModel.Name, setModel)
+	if err != nil {
+		t.Fatal("failed to set model", err)
+	}
+
+	itr := kv.NewPartitionIterator(ctx, store, (&kvtest.TestModel{}).ProtoReflect().Type(), modelPartitionKey)
+	require.NotNil(t, itr)
+	itr.Close()
+
+	itr = kv.NewPartitionIterator(ctx, store, (&kvtest.TestModel{}).ProtoReflect().Type(), modelPartitionKey)
```
Comment (eden-ohana) [inline pkg/kv/iterators_test.go:34]: duplication?

**Comment by N-o-Z** [comment]
- Resolved in #4118
